plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameUpdateFBTypeInterfaceChange.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Daniel Lindhuber
+ *     - initial API and implementation and/or initial documentation
+ *   Michael Oberlehner - outsourced inner class to own file
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.typemanagement.refactoring.rename;
+
+import java.text.MessageFormat;
+import java.util.EnumSet;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.commands.change.ChangeDataTypeCommand;
+import org.eclipse.fordiac.ide.model.commands.delete.DeleteInterfaceCommand;
+import org.eclipse.fordiac.ide.model.data.StructuredType;
+import org.eclipse.fordiac.ide.model.datatype.helper.IecTypes;
+import org.eclipse.fordiac.ide.model.helpers.PackageNameHelper;
+import org.eclipse.fordiac.ide.model.libraryElement.FBType;
+import org.eclipse.fordiac.ide.model.libraryElement.VarDeclaration;
+import org.eclipse.fordiac.ide.typemanagement.Messages;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.ConfigurableChange;
+import org.eclipse.gef.commands.Command;
+import org.eclipse.gef.commands.CompoundCommand;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+public class RenameUpdateFBTypeInterfaceChange extends ConfigurableChange<FBType> {
+
+	final StructuredType struct;
+
+	public RenameUpdateFBTypeInterfaceChange(final FBType type, final StructuredType struct) {
+		super(MessageFormat.format(Messages.DeleteFBTypeParticipant_Change_DeleteFBTypeInterface, type.getName(),
+				struct.getName()), EcoreUtil.getURI(type), FBType.class);
+		this.struct = struct;
+	}
+
+	@Override
+	public EnumSet<ChangeState> getAllowedChoices() {
+		return EnumSet.of(ChangeState.RENAME, ChangeState.CHANGE_TO_ANY, ChangeState.DELETE);
```

Comment:
Should we even offer the choices `CHANGE_TO_ANY` and `DELETE` when renaming?

Reply:
you are right, i think i will remove it unless someone explicitly wants it